src/main/java/org/folio/dao/PieceClaimingRepository.java

```diff
@@ -0,0 +1,50 @@
+package org.folio.dao;
+
+import io.vertx.core.Future;
+import io.vertx.sqlclient.SqlResult;
+import io.vertx.sqlclient.Tuple;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.rest.persist.Conn;
+
+import static org.folio.dao.audit.AuditOutboxEventsLogRepository.OUTBOX_TABLE_NAME;
+import static org.folio.rest.impl.PiecesAPI.PIECES_TABLE;
+import static org.folio.rest.impl.TitlesAPI.TITLES_TABLE;
+import static org.folio.rest.persist.PostgresClient.convertToPsqlStandard;
+
+public class PieceClaimingRepository {
+  private static final Logger log = LogManager.getLogger();
+  private static final String SYNTHETIC_USER_ID = "06c3485f-631c-427e-bade-5e763636c470";
+  private static final String UPDATE_STATEMENT = """
+    WITH updated AS (
+        UPDATE %1$s.%2$s as pieces
+        SET jsonb = pieces.jsonb || jsonb_build_object(
+            'receivingStatus', 'Late',
+            'statusUpdatedDate', to_char(clock_timestamp(),'YYYY-MM-DD"T"HH24:MI:SS.MSTZH:TZM')::text,
+            'metadata', pieces.jsonb -> 'metadata' || jsonb_build_object(
+                'updatedDate', to_char(clock_timestamp(),'YYYY-MM-DD"T"HH24:MI:SS.MSTZH:TZM')::text,
+                'updatedByUserId', $1))
+        FROM %1$s.%3$s as titles
+        WHERE pieces.titleid = titles.id
+        AND (titles.jsonb ->> 'claimingActive')::boolean = TRUE
```

Comment:
lets add index for field claimingActive, by default this flag would be disabled, so in this case with this index we can significanly reduce amount of records in WHERE cause

Reply:
And also I think we can swap and make (titles.jsonb ->> 'claimingActive')::boolean = TRUE as first statement in WHERE cause. In this case our 'claimingActive' index will filter more records and joins for piece table where claimingActive=false will not occur

Reply:
Thanks. Improved